### DISCUSSION TITLE
Output only when the outputclass is set, otherwise do not output any HTML

### DISCUSSION THREAD

**Discussion by fakhryilham** [dita-ot/dita-ot#3961] [Q&A]
I am trying to publish my DITA bookmap into single HTML files based on the outputclass set. So I want to only output when the outputclass is set, otherwise do not output any HTML

This is my build.xml file.
[build_dita2html5.zip](https://github.com/dita-ot/dita-ot/files/9043516/build_dita2html5.zip)

This still outputs HTML file of other DITA file that has no outputclass set.

I am trying to output only DITA files that has outputclass = 'article' set. I would also like to be able to set the outputclass = 'article' on the Bookmap level (map/topicref).

DITA profiling is not an option due to other reason in my environment. Therefore, I need to use the outputclass.

How can I just output the HTML file when only the outputclass is set?

**Comment by chrispy-snps** [comment]
In your discussion thread here:

https://dita-users.groups.io/g/main/message/46933

I described that I tried to do a similar thing a couple years ago - remove a `<draftintro>` topic from a bookmap when `args.draft == no`.

I found that profiling attributes have dedicated machinery to remove filtered topics from the .job.xml file in the temporary directory, which contains the list of topic/image files to be processed and published.

You can read about my struggles here:

#3663: Changes to .job.xml file are sometimes not detected and handled

https://groups.google.com/g/dita-ot-users/c/u6sGODXRCrw/m/41E-i3l4AgAJ

I eventually gave up and applied profiling conditions to my `<draftintro>` elements, in parallel with setting `args.draft` to keep/remove `<draft-comment>` elements within topics. It felt unnecessarily redundant but it worked.

I know you said that profiling attributes are not an option for you, but maybe some part of my struggles above might help you figure out a solution to your own situation.

**Comment by raducoravu** [comment]
Linking also to this stackoverflow post which is probably also by @fakhryilham : https://stackoverflow.com/questions/72781224/output-only-when-the-outputclass-is-set-otherwise-do-not-output-any-html

So what you want is not possible with the default DITA OT functionality.
I do not think you can achieve what you want just by XSLT customizations.
I see that this feature by @jelovirt was implemented some time ago:
https://github.com/dita-ot/dita-ot/pull/2784
allowing people to specify in the "dita-ot/config/configuration.properties" configuration file extra attributes to be interpreted as profiling attributes and thus to be filtered using a DITAVAL filter file. But I have not tested this myself.
If that does not work, the filter stage in the DITA OT is done with Java code, the DITA OT is an open source project so people can download it, make changes to its Java code and create their own custom DITA OT builds.
